components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/dao/CacheBackedScopeClaimMappingDAOImpl.java

```diff
@@ -135,4 +156,18 @@ private OIDCScopeClaimCacheEntry loadOIDCScopeClaims(int tenantId, OIDCScopeClai
         }
         return oidcScopeClaimCacheEntry;
     }
+
+    @Override
+    public ScopeDTO getScope(String scopeName, int tenantId) throws IdentityOAuth2Exception {
+
+        OIDCScopeClaimCacheEntry oidcScopeClaimCacheEntry = oidcScopeClaimCache.getScopeClaimMap(tenantId);
+        oidcScopeClaimCacheEntry = loadOIDCScopeClaims(tenantId, oidcScopeClaimCacheEntry);
+        ScopeDTO scopeDTO = null;
+        for (ScopeDTO scopeObj : oidcScopeClaimCacheEntry.getScopeClaimMapping()) {
+            if (scopeName.equals(scopeObj.getName())) {
+                scopeDTO = scopeObj;
```

Comment:
don't we need to break the loop from here?

Reply:
Fxied